src/pages/index.js

```diff
@@ -72,50 +88,166 @@ newCardPopup.setEventListeners();
   const name = profileData.title; //set name var to work with userInfo class
   const job = profileData.description; //set job var to work with userInfo class
   console.log("name:", name, "job:", job); //debug log
   user.setUserInfo({ name, job }); //set user information
+  api.updateUserInfo(profileData.title, profileData.description);
   const updatedUserInfo = user.getUserInfo(); //get updated information and store in a var
   console.log(updatedUserInfo); // debug log
   profilePopup.closePopup(); //close popup
+
+  setTimeout(() => {
+    profileModalSubmitButton.textContent = "Save";
+  }, 1000);
+}
+
+function handleAvatarUpdate(link) {
+  const url = link.url; //extract url from link object and set it to url value
+
+  avatarModalSubmitButton.textContent = "Saving...";
+
+  api
+    .updateProfilePic(url)
+    .then(() => {
+      user.setUserAvatar(url);
+    })
+    .catch((err) => {
+      console.error(`Error caught in HandleAvatarUpdate function ${err}`);
+    });
+
+  setTimeout(() => {
+    avatarModalSubmitButton.textContent = "Save";
+  }, 1000);
+  avatarPopup.closePopup();
 }
 
 function handleAddCardFormSubmit(newCardData) {
-  const name = newCardData.title; //set card name
-  const alt = newCardData.title; //set card alt text
-  const link = newCardData.url; //set card image link
-  console.log(name, alt, link, newCardData); //debug log
-
-  renderCard({ name, link, alt }); //call renderCard to append card to page
-  newCardPopup.closePopup(); //close popup
-  addNewCardValidator.resetForm(); //reset form and disable submit button
+  addNewCardSubmitButton.textContent = "Adding...";
+  api
+    .createNewCard(newCardData.title, newCardData.url)
+    .then((res) => {
+      /* console.log(res); */
+      renderCard(res);
+      newCardPopup.closePopup(); //close popup
+      addNewCardValidator.resetForm(); //reset form and disable submit button
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+
+  setTimeout(() => {
+    addNewCardSubmitButton.textContent = "Add";
+  }, 1000);
+}
+
+function handleCardLikes(cardData) {
+  /*   console.log(cardData);
+  console.log(cardData._like); */
+
+  if (!cardData._like) {
+    //if not liked
+    console.log(cardData._like);
+    api
+      .cardLike(cardData._id)
+      .then(() => {
+        /*    console.log(`Card ${cardData._id} liked`); */
+        cardData.updateLike(true); //call card.js like method
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else if (cardData._like) {
+    //if liked
+    api
+      .cardUnlike(cardData._id)
+      .then(() => {
+        cardData.updateLike(false);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
+
+function handleCardDelete(cardData) {
+  //pass in card data
+
+  deleteConfirm.openPopup(); //open delete modal
+  //close button functionality
+
+  console.log(cardData);
+  deleteModalButton.addEventListener("click", () => {
```

Comment:
Any modal handler should be added inside the class


So, it should be done inside `PopupWithConfirm`

Reply:
* When I delete 3 cards in a row, there are errors https://snipboard.io/MRAFsV.jpg 
*  Now all your cards (which you've tried to delete but changed your mind) are **deleted** all at once. Please try it yourself: add 3 cards in a row then try to delete them, but don't do it (cancel and close the confirmation popup) and then delete any other card - and all your previous cards are deleted at once (all 3 and the other one). It’s because of a **memory leak**. All listeners are gathered in one element and they are fired all together when you confirm deleting. You need either to **correctly remove** submit listeners or update the card’s data. Please, consult your mentor about that.